highlighters/main/main-highlighter.zsh

```diff
@@ -601,16 +601,19 @@ _zsh_highlight_main_highlighter_highlight_list()
       (( in_alias[1]-- ))
       # Remove leading 0 entries
       in_alias=($in_alias[$in_alias[(i)<1->],-1])
-      (){
-        local alias_name
-        for alias_name in ${(k)seen_alias[(R)<$#in_alias->]}; do
-          unset "seen_alias[$alias_name]"
-        done
-      }
       if (( $#in_alias == 0 )); then
         seen_alias=()
         # start_pos and end_pos are of the alias (previous $arg) here
         _zsh_highlight_main_add_region_highlight $start_pos $end_pos $alias_style
+      else
+        # We can't unset keys that contain special characters (] \ and some others).
+        # More details: https://www.zsh.org/mla/workers/2018/msg01073.html
```

Comment:
Nitpick, but how about linking to https://www.zsh.org/workers/43269?  Resolves to the same page, but leaving the X-Seq number in there is idiomatic, and will help people find the message in other archives if needed.